modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js

```diff
@@ -479,4 +518,115 @@ export default class NestedTabs extends Base {
 
 		return !! $tabTitleContainerElement && isTabTitleActive ? true : false;
 	}
+
+	changeScrollStatus( event ) {
+		const slider = this.elements.$headingContainer[ 0 ];
+
+		if ( 'mousedown' === event.type ) {
+			slider.classList.add( 'e-scroll' );
+			slider.dataset.pageX = event.pageX;
+		} else {
+			slider.classList.remove( 'e-scroll' );
+			slider.dataset.pageX = '';
+		}
+	}
+
+	isHorizontalScroll() {
+		const slider = this.elements.$headingContainer[ 0 ];
+
+		return slider.clientWidth < this.getChildrenWidth( slider.children ) && 'enable' === this.getHorizontalScrollSetting();
+	}
+
+	getChildrenWidth( children ) {
+		let totalWidth = 0;
+
+		const parentContainer = children[ 0 ].parentNode,
+			computedStyles = getComputedStyle( parentContainer ),
+			gap = parseFloat( computedStyles.gap ) || 0; // Get the gap value or default to 0 if it's not specified
+
+		for ( let i = 0; i < children.length; i++ ) {
+			totalWidth += children[ i ].offsetWidth + gap;
+		}
+
+		return totalWidth;
+	}
+
+	setHorizontalScrollAlignment( event = {} ) {
+		if ( ! this.elements ) {
+			return;
+		}
+
+		const slider = this.elements.$headingContainer[ 0 ];
+
+		if ( this.isHorizontalScroll() ) {
+			const tabsDirection = this.getTabsDirection();
+
+			this.initialScrollPosition( slider, tabsDirection );
+		} else {
+			slider.style.setProperty( '--n-tabs-heading-justify-content', '' );
+		}
+	}
+
+	getTabsDirection() {
+		const currentDevice = elementorFrontend.getCurrentDeviceMode(),
+			tabsDirection = elementorFrontend.utils.controls.getResponsiveControlValue( this.getElementSettings(), 'tabs_justify_horizontal', '', currentDevice );
+
+		return tabsDirection;
+	}
+
+	initialScrollPosition( slider, tabsDirection ) {
+		switch ( tabsDirection ) {
+			// To cover defaults and center
+			case undefined:
+			case '':
+			case 'center':
+				slider.style.setProperty( '--n-tabs-heading-justify-content', 'start' );
+				slider.scrollLeft = ( this.getChildrenWidth( slider.children ) - slider.offsetWidth ) / 2;
```

Comment:
Why do we subtract `slider.offsetWidth`?

Reply:
@ArmanPurtoian This part isn't clear to me.
Doesn't `slider.style.setProperty( '--n-tabs-heading-justify-content', 'start' );` overwrite center styling?
I am curious about this aspect.